### PR TITLE
Adding Cost Optimization

### PR DESCRIPTION
**Pull request by senkinnar**
*Issue #, if available:*

*Description of changes:*
Add Cost Optimization best practices with EC2 Spot section.
- Node Decommissioning, PVC Reuse, Scaling 
Removed Spot instances resiliency section

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/cost-optimization/docs/cost-optimization.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+## EC2 Spot Best Practices
+
+Amazon EMR on Amazon EKS enables you to submit Apache Spark jobs on demand on Amazon Elastic Kubernetes Service (EKS) without provisioning dedicated EMR clusters. With EMR on EKS, you can consolidate analytical workloads with your other Kubernetes-based applications on the same Amazon EKS cluster to improve resource utilization and simplify infrastructure management. Cost Optimization of the underlying infrastructure is often the key
+requirement for our customers, and this can be achieved by using [Amazon EC2 Spot Instances](https://aws.amazon.com/ec2/spot/). Spot Instances are spare EC2 capacity and is available at up to 90% discount compared to On-Demand Instance prices. If EC2 needs capacity back for On-Demand Instance usage, Spot Instances can be interrupted. Handling interruptions to build resilient workloads is simple and there are best practices to manage interruption by automation or AWS services like EKS. 
+
+This document will describe how to think about EC2 Spot and its best practices when applying to EMR on EKS service. We will cover Spark features related to EC2 Spot when you run EMR on EKS jobs on EKS cluster.
```
Comment (dalbhanj) [inline content/cost-optimization/docs/cost-optimization.md]: This document will describe how to architect with EC2 spot best practices and apply to EMR on EKS jobs. We will also cover Spark features related to EC2 Spot when you run EMR on EKS jobs

File: content/spot-instances-resiliency/docs/copying-shuffle-data.md [hunk 2]
```diff
@@ -1,97 +0,0 @@
-# **Spot instances interruption resiliency by copying shuffle data**
```
Comment (dalbhanj) [inline content/spot-instances-resiliency/docs/copying-shuffle-data.md:1]: I feel some of this content is useful. For ex, https://aws.github.io/aws-emr-containers-best-practices/spot-instances-resiliency/docs/copying-shuffle-data/#how-does-it-work. and other sections such as "Observed Behavior" and "Node decommissioning with fallback storage". 

Can we create separate doc focusing on decommissioning feature and merge this content?

**Review by dalbhanj — CHANGES_REQUESTED** [review]

**Comment by dalbhanj** [comment]
@kmsiddh can you please review and provide feedback?

File: content/cost-optimization/docs/cost-optimization.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+**Node Decommissioning:**
+
+Node decommissioning is a Spark feature that enables the removal of an executor gracefully, by preserving its state before removing it and not scheduling any new jobs on it. This feature is particularly useful when the Spark executors are running on Spot instances, and the Spark executor node is interrupted via a ‘rebalance recommendation’ or ‘instance termination’ notice to reclaim the instance. 
+
+Node decommission begins when a Spark executor node receives a Spot Interruption Notice or Spot Rebalance Recommendation signal. The executor node immediately starts the process of decommissioning by sending a message to the Spark driver. The driver will identify the RDD/Shuffle files that it needs to migrate off the executor node in question, and will try to identify another Executor node which can take over the execution. If an executor is identified, the RDD/Shuffle files are copied to the new executor and the job execution continues on the new executor. If all the executors are busy, the RDD/Shuffle files are copied to an external storage.
```
Comment (kmsiddh) [inline content/cost-optimization/docs/cost-optimization.md:85]: > If all the executors are busy, the RDD/Shuffle files are copied to an external storage.

To be specific, are we talking about `spark.storage.decommission.fallbackStorage.path` config? Last I had seen there was an issue in this feature. We were seeing the error `java.io.IOException: Stream is corrupted, net.jpountz.lz4.LZ4Exception: Error decoding offset <offset_id> of input buffer`. If we have tested the config and it works, can we call out the EMR release that it works from?

Comment (senkinnar) [reply content/cost-optimization/docs/cost-optimization.md:85]: Yes this is working now. I will update the exact version after some testing.

File: content/spot-instances-resiliency/docs/copying-shuffle-data.md (hunk 2 above)
Comment (senkinnar) [reply content/spot-instances-resiliency/docs/copying-shuffle-data.md:1]: This has been addressed as discussed, please review

File: content/cost-optimization/docs/cost-optimization.md (hunk 1 above)
Comment (senkinnar) [reply content/cost-optimization/docs/cost-optimization.md]: Added

**Comment by dalbhanj** [comment]
lgtm

**Pull request merged**